tests/test_pglogical_do_blocks.py

```diff
@@ -0,0 +1,111 @@
+"""Test pglogical nested DO block patterns."""
+
+from pgrubic import core
+
+
+def test_pglogical_nested_do_block_current_behavior(linter: core.Linter):
+    """Test current behavior with pglogical nested DO blocks."""
+    
+    # This represents the realistic production pattern
+    pglogical_sql = '''
+SELECT pglogical.replicate_ddl_command(
+$$
+    DO
```

Comment:
I can think of few ideas for this but what of cases when it is reversed. A DO block containing a function call with another DO block.

Reply:
The best pgrubic can do is lint first level of inline migrations. Going deeper will easily complicate things.

Reply:
This also holds for inline `create function` (that has several other statements) in a do block.